packages/wdio-runner/src/index.ts

```diff
@@ -411,6 +411,16 @@ export default class Runner extends EventEmitter {
         failures: number,
         retries: number
     ) {
+        this._reporter.emit('runner:start', {
```

Comment:
Can we only do this if there was no `runner:start` in the first place. Also can we add a comment to ensure people understand what is behind this?

Reply:
SOLUTION 1:  

```
     /**
         * In case of initialisation failed, the sessionId is undefined and the runner:start is not triggered.
         * So, to be able to perform the runner:end into the reporters, we need to trigger the runner:start
         */
        if (!global.browser.sessionId) {
            this._reporter.emit('runner:start', {
                cid: this._cid,
                specs: this._specs,
                config: this._config,
                isMultiremote: this._isMultiremote,
                instanceOptions: {},
                capabilities: {...this._configParser.getCapabilities()},
                retry: this._specFileRetryAttempts
            });
        }
```

Not 100% sure about checking the global.browser.sessionId, as i the endSession you check if the sessionId exists:
https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-runner/src/index.ts#L434.
So, maybe in specific cases the sessionId is not defined... don't know.


SOLUTION 2:

If you prefer I can just create a `_runnerStartWasCalled` like the `_sigintWasCalled`: https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-runner/src/index.ts#L75 like: 
```
private _runnerStartWasCalled = false

// Just before the runner:started:
this._runnerStartWasCalled = true


In the shutdown:
/**
         * In case of initialisation failed, the runner:start is not triggered.
         * So, to be able to perform the runner:end into the reporters, we need to trigger the runner:start
         */
if (!_runnerStartWasCalled) {
            this._reporter.emit('runner:start', {
                cid: this._cid,
                specs: this._specs,
                config: this._config,
                isMultiremote: this._isMultiremote,
                instanceOptions: {},
                capabilities: {...this._configParser.getCapabilities()},
                retry: this._specFileRetryAttempts
            });
        }
```

Reply:
Solution 1 makes sense to me and seems readable enough.